packages/flutter/lib/src/animation/animation.dart

```diff
@@ -60,33 +60,93 @@ typedef AnimationStatusListener = void Function(AnimationStatus status);
+/// forward, backward, or stopped at the end or the beginning.
+/// The path may double back on itself
+/// (e.g., if the animation uses a curve that bounces),
+/// so even when the animation is conceptually moving forward
+/// the value might not change monotonically.
 ///
-/// Animations also let other objects listen for changes to either their value
-/// or their status. These callbacks are called during the "animation" phase of
+/// Consumers of the animation can listen for changes to either the value
+/// or the status, with [addListener] and [addStatusListener].
+/// The listener callbacks are called during the "animation" phase of
 /// the pipeline, just prior to rebuilding widgets.
 ///
-/// To create a new animation that you can run forward and backward, consider
-/// using [AnimationController].
+/// An animation may move forward or backward on its own as time passes
+/// (like the opacity of a button that fades over a fixed duration
+/// once the user touches it),
+/// or it may be driven by the user
+/// (like the position of a slider that the user can drag back and forth),
+/// or it may do both
+/// (like a switch that snaps into place when releaseed,
+/// or a [Dismissible] that responds to drag and fling gestures, etc.).
+/// The behavior is normally controlled by method calls on
+/// some underlying [AnimationController],
+/// for example by a [State] that created the [AnimationController].
+/// When an animation is actively animating, it typically updates on
+/// each frame, driven by a [Ticker].
+///
+/// ## Using animations
+///
+/// For simple animation effects, consider using one of the
+/// [ImplicitlyAnimatedWidget] subclasses,
+/// like [AnimatedScale], [AnimatedOpacity], and many others.
+/// When an [ImplicitlyAnimatedWidget] suffices, there is
+/// no need to work with [Animation], [AnimationController], or [Tween],
+/// or even to have a [State] to manage the animation (though all of those are
+/// involved in the implementation of the [ImplicitlyAnimatedWidget] itself).
+///
+/// Otherwise, typically an animation originates with an [AnimationController]
+/// (which is itself an [Animation<double>])
+/// created by a [State] that implements [TickerProvider],
+/// and then further animations may be derived from it using
+/// e.g. [Tween] or [CurvedAnimation].
+/// The animations might be used to configure an [AnimatedWidget]
+/// (using one of its many subclasses like [FadeTransition]),
+/// or their values might be used directly.
+///
+/// For example, the [AnimationController] may represent
+/// the abstract progress of the animation from 0.0 to 1.0;
+/// then a [CurvedAnimation] might apply an easing curve;
+/// and a [SizeTween] and [ColorTween] might each be applied to that
+/// to produce an [Animation<Size>] and an [Animation<Color>] that control
+/// a widget shrinking and changing color as the animation proceeds.
+///
+/// Because the [Animation] keeps the same identity as the animation proceeds,
```

Comment:
This is an excellent tip. Do you think it deserves its own section?

```suggestion
/// ## Performance
///
/// Because the [Animation] keeps the same identity as the animation proceeds,
```

Reply:
Sure, good idea.

I expanded the heading slightly to "Performance considerations", following other examples in the codebase.